docs/develop/dialog-best-practices.md

```diff
@@ -48,6 +48,67 @@ If, for any reason, you want to turn off this feature, then your code must opt o
 
 Office automatically adds a query parameter called `_host_info` to the URL that is passed to `displayDialogAsync`. It is appended after your custom query parameters, if any. It is not appended to any subsequent URLs that the dialog box navigates to. Microsoft may change the content of this value, or remove it entirely, so your code should not read it. The same value is added to the dialog box's session storage (that is, the [Window.sessionStorage](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage) property). Again, *your code should neither read nor write to this value*.
 
+### Opening another dialog immediately after closing one
+
+You cannot have more than one dialog open from a given host page, so your code should call [Dialog.close](/javascript/api/office/office.dialog#close__) on an open dialog before it calls `displayDialogAsync` to open another dialog. The `close` method is asynchronous. For this reason, if you call `displayDialogAsync` immediately after a call of `close`, the first dialog may not have completely closed when Office attempts to open the second. If that happens, Office will return a [12007](dialog-handle-errors-events.md#12007) error: "The operation failed because this add-in already has an active dialog."
+
+The `close` method doesn't accept a callback parameter, and it doesn't return a Promise object so it cannot be awaited with either the `await` keyword or with a `then` method. For this reason, we suggest the following technique when you need to open a new dialog immediately after closing a dialog: encapsulate the code to open the new dialog in a method and design the method to recursively call itself if the call of `displayDialogAsync` returns `12007`. The following is a simple example.
+
+```javascript
+function openFirstDialog() {
+  Office.context.ui.displayDialogAsync("https://MyDomain/firstDialog.html", { width: 50, height: 50},
+     (result) => {
+      if(result.status === Office.AsyncResultStatus.Succeeded) {
+        const dialog = result.value;
+        dialog.close();
+        openSecondDialog();
+      }
+      else {
+         // Handle errors
+      }
+    }
+  );
+}
+ 
+function openSecondDialog() {
+  Office.context.ui.displayDialogAsync("https://MyDomain/secondDialog.html", { width: 50, height: 50},
+    (result) => {
+      if(result.status === Office.AsyncResultStatus.Failed) {
+        if (result.error.code === 12007) {
+          openSecondDialog(); // Recursive call
+        }
+        else {
+         // Handle other errors
+        }
+      }
+    }
+  );
+}
+```
+
+Alternatively, you could force the code to pause before it tries to open the second dialog by using the [setTimeout](https://www.w3schools.com/jsref/met_win_settimeout.asp) method. The following is an example.
```

Comment:
```suggestion
// Alternatively, you could force the code to pause before it tries to open the second dialog by using the [setTimeout](https://www.w3schools.com/jsref/met_win_settimeout.asp) method. The following is an example.
```
I think you need something to indicate this is a comment

Reply:
Actually, maybe make this regular text and separate the below into it's own chunk of code?

Reply:
That's the way it is already: regular text between two code blocks/